src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/behandlingsresultat/BehandlingsresultatOpphørUtils.kt

```diff
@@ -67,17 +67,22 @@ object BehandlingsresultatOpphørUtils {
         val meldtBarnehageplassPåAlleBarnMedLøpendeAndeler =
             alleBarnMedLøpendeAndeler.isNotEmpty() &&
                 alleBarnMedLøpendeAndeler.all { barn ->
-                    personResultater.any { personresultat ->
-                        personresultat.aktør == barn.aktør &&
-                            personresultat.vilkårResultater.any { vilkårResultat ->
-                                vilkårResultat.harFullBarnehageplass()
-                            }
-                    }
+                    barnHarBarnehageplass(personResultater, barn)
                 }
 
         return meldtBarnehageplassPåAlleBarnMedLøpendeAndeler
     }
 
+    private fun barnHarBarnehageplass(
+        personResultater: List<PersonResultat>,
+        barn: PersonResultat,
+    ) = personResultater.any { personresultat ->
+        personresultat.aktør == barn.aktør &&
+            personresultat.vilkårResultater.any { vilkårResultat ->
+                vilkårResultat.harFullBarnehageplass()
+            }
+    }
```

Comment:
Kanskje litt utenfor oppgaven, men hvorfor kan vi ikke bare se på `barn: PersonResultat`? Og hvorfor kaller vi et PersonResultat for barn? Er dette barnet sitt person resultat? Kan vi i så fall leste om det har barnehageplass direkte fra det?

Reply:
burde alleBarnMedLøpendeAndeler renames til `personResultaterForBarnMedLøpendeAndeler`?

Reply:
Enig, skal være en-til-en forhold mellom personresultat og aktør så gir ikke mening å iterere gjennom det på denne måten.